### PR TITLE
RateMyProfessor Web Scraping + Filter

### PR DESCRIPTION
**Pull request by BakerWJ**
# Description

Implements JSoup to scrape RateMyProfessor to assess the quality of the profs in each section. This score is then used by the RMPFilter, which can enforce a minimum allowed score for a professor.

## Type of change

- [x] New feature 
- [x] Breaking change 
- [x] This change requires a documentation update

# How Has This Been Tested?

No tests added yet, but I have changed the timeout of certain tests to make sure that they still pass, even though the scraper is being used.

# Does this PR follow the rubric?

No design patterns used.

# Checklist:

- [x] I ran `./gradlew spotlessApply`
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I ran `./gradlew build` and the gradle build succeeded
- [x] I have not violated any design principles

# Other Comments:

Other comments that should be made clear.

### CODE REVIEW

File: src/main/java/workers/ProfessorRatingScraper.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import java.io.IOException;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.*;
+import java.util.HashMap;
+import java.util.Map;
+
+class ProfessorRatingScraper {
+    private static Map<String, Double> previousScores = new HashMap<String, Double>();
+
+    public static double getRating(String firstName, String lastName) {
```
Comment (kenneth-miura) [inline src/main/java/workers/ProfessorRatingScraper.java:12]: chris wants us to use static less to show we understand OOP. Can you refactor this so it's not static?

**Review by kenneth-miura — CHANGES_REQUESTED** [review]
looks good. How much does running the RateMyProfessor webscraping for every section affect runtime? Is it noticeable when u run through CLI?

File: src/main/java/workers/ProfessorRatingScraper.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            html = Jsoup.connect(
+                    "https://www.ratemyprofessors.com/search/teachers?query="
+                            + firstName
+                            + " "
+                            + lastName
+                            + "&sid=U2Nob29sLTE0ODQ=")
+                    .get()
+                    .html();
+        } catch (IOException e) {
+            return 2.5;
```
Comment (EclipseIsDead) [inline src/main/java/workers/ProfessorRatingScraper.java:27]: This is a big no no, throw the error. I see that we need a return but you could refactor it to just throw the error instead w/out having to return 2.5. Up to you, at the very least throw the error.

File: src/test/java/filtersTests/CourseExclusionFilterTest.java [hunk 3]
```diff
@@ -35,7 +35,7 @@ public void filterFail() {
         assertFalse(filter1.checkSchedule(schedule));
     }
 
-    @Test(timeout = 1000)
+    @Test(timeout = 5000)
```
Comment (EclipseIsDead) [inline src/test/java/filtersTests/CourseExclusionFilterTest.java:38]: Why did our runtimes like triple?

**Review by EclipseIsDead — CHANGES_REQUESTED** [review]
Good stuff, 2 things:
- Add tests. If you don't want to/can't then I will just lmk
- Why did you need to increase the timeouts for tests

**Pull request merged**